index.js

```diff
@@ -53,15 +54,21 @@ function readPackageJSON (encoding) {
 function filterConfig (configTmpl, pkg) {
     // create a config object from the config template
     // replace {key} with the key's value in package.json
+    // replace ${key} with eval(key)
     var obj = extend({}, configTmpl);
+    var replacibleKeys = {
+        env: process.env,
+        package: pkg
+    };
     Object.keys(obj).forEach(function (key) {
         var value = obj[key];
         if (typeof value != 'string') { return; }
 
-        obj[key] = value.replace(/{([^}]+)}/g, function (org, key) {
-            if (pkg[key] === undefined) { return org; }
-            return pkg[key];
-        });
+        obj[key] = Mustache.render(value, replacibleKeys)
+            .replace(/{([^}]+)}/g, function (org, key) {
```

Comment:
@skyway777 have you tested if single curly braces could work with mustache? Maybe we could remove the old replacing mechanism entirely?!

Reply:
no, it doesn't. 
No one template engine supports it, because it can be part of text or code.